src/template.rs

```diff
@@ -91,8 +102,59 @@ impl Template {
         self
     }
 
-    pub fn render(self) -> String {
-        let (year_repl, author_repl, ident_repl) = if self.spdx_template {
+    pub fn outdated_license_pattern(&self, commenter: &Box<dyn Comment>, columns: Option<usize>) -> Regex {
+        self.build_year_varying_regex(&mut self.content.clone(), commenter, columns, false)
+    }
+
+    pub fn outdated_license_trimmed_pattern(&self, commenter: &Box<dyn Comment>, columns: Option<usize>) -> Regex {
+        self.build_year_varying_regex(&mut self.content.clone(), commenter, columns, true)
+    }
+
+    pub fn render(&self) -> String {
+        let (year_repl, author_repl, ident_repl) = self.replacement_tokens();
+        let template = self.unwrap_license_headers(&mut self.content.clone());
+
+        // Perform our substitutions
+        template
+            .replace(year_repl, &self.context.get_year())
+            .replace(author_repl, &self.context.get_authors())
+            .replace(ident_repl, &self.context.ident)
+    }
+
+    fn build_year_varying_regex(&self, content: &mut str, commenter: &Box<dyn Comment>, columns: Option<usize>, trim_trailing: bool) -> Regex {
+        let (year_repl, author_repl, ident_repl) = self.replacement_tokens();
+        let mut rendered = commenter.comment(
+            &(self.unwrap_license_headers(content)
+                .replace(year_repl, INTERMEDIATE_YEAR_TOKEN)
+                .replace(author_repl, &self.context.get_authors())
+                .replace(ident_repl, &self.context.ident)),
+            columns,
+        );
+
+        if trim_trailing {
+            rendered = trim_trailing_whitespace(&rendered).to_string()
+        }
+
+        let escaped: Vec<_> = rendered
```

Comment:
This needs some comments as it is very dense. I understand a bit how this works because of your excellent comment above but reading this code it's really unclear to me what the actual value of `escaped` is supposed to be. Doubly so when I look at L145 and try to workout what the regex would be.